homeassistant/components/template/__init__.py

```diff
@@ -29,13 +30,61 @@
 
 from .const import CONF_MAX, CONF_MIN, CONF_STEP, CONF_TRIGGER, DOMAIN, PLATFORMS
 from .coordinator import TriggerUpdateCoordinator
+from .helpers import async_get_blueprints
+from .template_entity import TemplateEntity
 
 _LOGGER = logging.getLogger(__name__)
 DATA_COORDINATORS: HassKey[list[TriggerUpdateCoordinator]] = HassKey(DOMAIN)
 
+DATA_COMPONENT: HassKey[EntityComponent[TemplateEntity]] = HassKey(DOMAIN)
+
+
+@callback
+def templates_with_blueprint(hass: HomeAssistant, blueprint_path: str) -> list[str]:
+    """Return all templates that reference the blueprint."""
+    return [
+        template_entity.entity_id
+        for platform in PLATFORMS
+        if platform in hass.data
+        for template_entity in hass.data[platform].entities
```

Comment:
We're not allowed to access another integration's `hass.data` items directly. We can use this helper to get all platforms of the template integration.

https://github.com/home-assistant/core/blob/9921a67a05bdb1b1a17b08b1bebbde7f90b250c3/homeassistant/helpers/entity_platform.py#L1063-L1066

The `EntityPlatform` has a dict with all its entities in the `entities` attribute.